__tests__/api/chat/channels.test.js

```diff
@@ -20,31 +46,88 @@ describe('GET /chat/channels/signed', () => {
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      3,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 60)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      4,
+      ...__generateQueryChannelsParamsCall(user.user_id, 10, 90)
+    );
+  });
+
+  test('should return 200 OK with list of channels with offset only', async () => {
+    const user = await createUser();
+
+    const response = await request(app)
+      .get('/api/v1/chat/channels/signed?offset=10')
+      .set({Authorization: `Bearer valid-token`})
+      .expect(200);
+
+    expect(response.body).toEqual(__getSignedChannelListBodyExpectation);
+    expectGetSignedChannelListFlow(user.user_id);
+
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      1,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 10)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      2,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 40)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      3,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 70)
     );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      4,
+      ...__generateQueryChannelsParamsCall(user.user_id, 10, 100)
+    );
+  });
+
+  test('should return 200 OK with list of channels with limit only', async () => {
+    const user = await createUser();
+
+    const response = await request(app)
+      .get('/api/v1/chat/channels/signed?limit=10')
+      .set({Authorization: `Bearer valid-token`})
+      .expect(200);
 
-    expect(jwt.verify).toHaveBeenCalled();
-    expect(StreamChat.getInstance).toHaveBeenCalled();
-    expect(StreamChat._instance).toBeInstanceOf(StreamChat);
-    expect(StreamChat._instance.connectUser).toHaveBeenCalledWith(
-      {id: user.user_id},
-      'valid-token'
+    expect(response.body).toEqual(__getSignedChannelListBodyExpectation);
+    expectGetSignedChannelListFlow(user.user_id);
+
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      1,
+      ...__generateQueryChannelsParamsCall(user.user_id, 10, 0)
     );
-    expect(StreamChat._instance.queryChannels).toHaveBeenCalledWith(
-      {members: {$in: [user.user_id]}},
-      [{last_message_at: -1}]
+  });
+
+  test('should return 200 OK with list of channels with limit and offset', async () => {
+    const user = await createUser();
+
+    const response = await request(app)
+      .get('/api/v1/chat/channels/signed?limit=10&offset=10')
+      .set({Authorization: `Bearer valid-token`})
+      .expect(200);
+
+    expect(response.body).toEqual(__getSignedChannelListBodyExpectation);
+    expectGetSignedChannelListFlow(user.user_id);
+
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      1,
+      ...__generateQueryChannelsParamsCall(user.user_id, 10, 10)
     );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Same as above, avoid making assumptions about the internal implementation of the `queryChannels` method.


<!-- This is an auto-generated comment by OSS CodeRabbit -->